VHAR-6946 POT Kulutuskerros hypyt

solita-staskila opened the pull request:
Korosta hypyt päällystyksessä (max 50m)

solita-staskila committed:
Tunnista sekä korosta POT kulutuskerros hypyt

solita-staskila committed:
Hyppy alert, kokonaispituus html

solita-staskila committed:
Hypyt OK tyylit

solita-staskila committed:
Laske kulutuskerroksen hyppyjen määrä

solita-staskila committed:
Laske teiden kokonaispituus

solita-staskila committed:
Käsittele hyppyjen määrä

solita-staskila committed:
Näytä header vain kulutuskerroksessa

solita-staskila committed:
Korosta molemmat aet/let jos tarve, korosta myös ei-muokatavat elementit

solita-staskila commented:
Olettaisin että hyppyjä lasketaan vain silloin kuin ollaan samalla ajoradalla(?) tämä täytynee korjata

solita-staskila committed:
Tunnista hypyt vain samalla tiellä sekä ajoradalla

solita-staskila committed:
Korjaa kommentit

rckopteri commented on dev-resources/less/grid.less:
```diff
@@ -883,3 +883,39 @@ span.ajax-loader-pisteet {
+  border-bottom: 5px solid rgba(0,0,0,0.0) !important;
+}
+
+.kulutus-hyppy-info {
```
classin nimi viittaa siihen, että sitä käytetään vain yhdessä paikassa. Siirtäisin sen siis pois yleisesta grid.lessistä esim paallystys_ja_paikkaus.less tiedostoon. Sama noissa muissa, jotka eivät ole yleisiä määritteitä. Tai jos ovat, niin sitten nimi pitää vaihtaa sellaiseksi, ettei se liity vain kulutukseen.

rckopteri commented on src/cljs/harja/ui/grid/muokkaus.cljs:
```diff
@@ -20,9 +20,21 @@
 (defn- muokkauspaneeli [{:keys [otsikko voi-muokata? voi-kumota? muokatut virheet varoitukset huomautukset
                                 skeema peru! voi-lisata? ohjaus uusi-id opts paneelikomponentit historia
-                                virhe-viesti custom-toiminto]}]
+                                virhe-viesti custom-toiminto nayta-hypyt? hyppyjen-maara]}]
```
Onko "hypyt" sellaisia, mitä voi muissakin taulukoissa lisäillä? Vai pitäisikö nimetä jotenkin eri tavalla?

solita-staskila replied:
Vaihdoin parametriksi "custom-yla-panel" ja siirsin html generoinnin kokonaan päällystysten puolelle, onko OK? 

https://github.com/finnishtransportagency/harja/pull/2973/commits/53f66681e04b83b0296da4d1a339196ffd0dc423

rckopteri commented on src/cljs/harja/ui/grid/muokkaus.cljs:
```diff
@@ -20,9 +20,21 @@
+     (when nayta-hypyt?
+       [:div.kulutus-hyppy-info
+        [:div.kulutus-hyppy-ikoni-ok (ikonit/harja-icon-status-completed)]
+        [:div.kulutus-hyppy-teksti "Kulutuskerros on yhtenäinen (ei hyppyjä)"]]))
```
Kulutuskerroksesta puhuminen pitänee siirtää pois yleisestä komponentista.

rckopteri commented on src/cljs/harja/ui/kentat.cljs:
```diff
@@ -317,12 +317,14 @@
+                             input-luokka (str " " input-luokka)
+                             veda-oikealle? (str " veda-oikealle")
+                             ;; Piilota inputin tausta sekä borderit jotta parent elementin (td) korostus näkyy
+                             korosta-hyppy? (str " lapinakyva-tausta-muokkaus-input"))
```
Tästäkin sanoisin, että vähän liian spesifi parametri kaikille numero -tyyppisille kentille.

jarnovayrynen replied:
Joo samaa mieltä, tuo hypyn korostaminen ei kuulu tänne geneeriseen UI-komponenttiin. Otanpa haaran auki ja katselen vaihtoehtoja..

solita-staskila replied:
Kaikki päällystys koodi vedetty irti muokkaus sekä kentat cljs, mutta näihin piti tehdä kuitenkin muutoksia jotta halutut toiminnot näkyy oikein. Kaikki nimet vaihdettu geneeriseksi. 

https://github.com/finnishtransportagency/harja/pull/2973/commits/3e4ecd4fe8b44aeaeabae7d23528da51cc2c5c1f

rckopteri commented on src/cljs/harja/views/urakka/pot2/paallystekerros.cljs:
```diff
@@ -55,123 +55,157 @@
+                                                                         (map (juxt key (comp :pituus val)))
+                                                                         (get tr-osien-pituudet (:tr-numero (second data))))
+                                                   (second data))]
+                                 (+ acc pituus))) 0 @kohdeosat-atom)]
```
0 @kohdeosat-atom - parametrit voisi erottaa omalle rivilleen, niin lukeminen helpottuisi :)

solita-staskila replied:
Tehdään näin :+1:

solita-staskila committed:
Siirrä päällystys html generointi päällystykseen

solita-staskila commented:
Kun siirrytään kaistalta toiselle, se ei ole hyppy

solita-staskila committed:
Laske päällystys hypyt palvelimen puolella

solita-staskila committed:
Poista muokkaus gridistä päällystys koodit

solita-staskila committed:
Merge branch 'develop' into VHAR-6946_kulutuskerros_hypyt

solita-staskila committed:
Käsittele rivien korostus oikein muokkausgridissä

solita-staskila committed:
Tyhjennä hypyt kun lomaketta muokataan

solita-staskila committed:
Testaa POT tallennuksen hypyt

solita-staskila committed:
Korjaa molempien sarakkeiden korostus

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-6946_kulutuskerros_hypyt

solita-staskila committed:
Korjaa hyppyjen laskenta & korostus

jarnovayrynen commented on src/cljs/harja/views/urakka/pot2/pot2_lomake.cljs:
```diff
@@ -284,6 +284,7 @@
     (komp/luo
       (komp/lippu pot2-tiedot/pot2-nakymassa?)
       (komp/sisaan (fn [this]
+                     (e! (pot2-tiedot/->KulutuskerrosMuokattu false))
```
Käsittääkseni aina kun tullaan näkymään, pitäisi lähtötilanteen olla se ettei mitään ole muokattu? Eli ei vaadi erillistä asettamista?

solita-staskila replied:
Koska tuo tieto on (defonce), silloin kun muokataan lomaketta, arvo muuttuu trueksi eikä se muutu enää ikinä falseksi jos ladataan lomake uudelleen ellei sitä jossain aseteta. Mutta otin tuon pois ja siirsin tämän -> HaePot2TiedotOnnistui

jarnovayrynen commented on src/clj/harja/palvelin/palvelut/yllapitokohteet/paallystys.clj:
```diff
@@ -242,22 +242,158 @@
+                   (> rivi-aet edellinen-rivi-let)
+                   ;; Onko hyppy X metriä tai alle?
+                   (< (- rivi-aet edellinen-rivi-let) hypyn-metriraja))
+                 (merge rivi {:tr-korosta-aet? true})
```
Nimet ovat vaikeita, mutta mietin olisiko tässä tuo "korosta aet"  liiaksi presentaatioon keskittyvä termi? Olisiko tarkempaa puhua ihan :aet-hyppy? true? Eli palvelin ottaa kantaa vain ominaisuuteen, ei presentaatioon. UI-kerros sitten tekee tuolla hyppytiedolla mitä haluaa - alkuvaiheessa korostaa sitä kenties eri värillä jne. mutta ei naulita palvelinkoodiin sitä, miten jokin data esitetään.

solita-staskila replied:
Joo, oikeassa olet. 
Nimesin uudelleen -> :aet-hyppy? / :let-hyppy?

solita-staskila replied:
Joo, oikeassa olet.
Nimesin uudelleen -> :aet-hyppy? / :let-hyppy?

solita-staskila committed:
Korjaa muuttujien nimiä

solita-staskila committed:
Älä kopioi hyppytietoja uudelle riville

jarnovayrynen committed:
Lisää gridiin td-luokka-fn ja vie hypyn luokka sitä kautta

-Lisää muokkaus.gridiin td-luokka-fn, joka käyttää rivin tietoja. Vie tarvittaessa datapohjaisesti hyppyyn liittyvä luokka gridille.
-Näytä input-kentässäkin hypyn korostukseen liittyviä tyylejä sisäkkäistä LESS-selekoria hyödyntäen, kun luokka korostettu-sarake on päällä.

jarnovayrynen committed:
Vie kulutuskerros-muokattu? päällystysilmoituksen lomakedataan

Passaa kulutuskerros-muokattu? parametrina päällystekerrosnäkymälle

jarnovayrynen committed:
Uudelleennimeä td-luokka-fn --> solun-luokka-fn

Nimetään muokkausgridin uusi ominaisuus solun-luokaksi, koska perus-gridissä vastaavankaltainen toiminto sillä nimellä.

solita-staskila committed:
Pyyhi korostukset kun lomaketta muokataan

solita-staskila committed:
Siivoa nestattu let

solita-staskila committed:
Merge branch 'develop' into VHAR-6946_kulutuskerros_hypyt
